xmtp_mls/src/xmtp_openmls_provider.rs

```diff
@@ -4,25 +4,68 @@ use openmls_traits::OpenMlsProvider;
 use crate::storage::{db_connection::DbConnection, sql_key_store::SqlKeyStore};
 
 #[derive(Debug)]
-pub struct XmtpOpenMlsProvider<'a> {
+pub struct XmtpOpenMlsProvider {
+    crypto: RustCrypto,
+    key_store: SqlKeyStore<'static>,
+}
+
+pub struct XmtpOpenMlsProviderRef<'a> {
```

Comment:
Out of curiosity, why do we need this separate type?

Reply:
I'm still working through the compile errors, so I might do away with this in the end. But, it 's useful when we want to work with `OpenMlsProvider` and only have a `&DbConnection` (maybe we're deep in a codepath).

Reply:
Ultimately removed it